docs/deploying-guardrails.md

```diff
@@ -0,0 +1,90 @@
+
+So far, we've talked a lot about reducing the resources necessary to run Guardrails in production.  Another important factor we considered when shifting to the new client/server paradigm is how this pattern enables better scaling practices.  Since the Guardrails API is now separate and distinct from your application code, you can scale both separately and according to their own needs.  This means that if your client application needs to scale it can do so without accounting for the additional resources required for validation.  Likewise, if the validation traffic is the limiting factor, you client application can stay small on fewer instances while the Guardrails API scales out to meet the demand.  You're also not limited to only one Guardrails API deployable giving you the option to scale more heavily utilized use-cases independently of those less frequently used.
+
+
+## Considerations for Productionization
+As previously mentioned, the Guardrails API is currently a simple Flask application.  This means you'll want a WSGI server to serve the application for a production environment.  There are many options out there and we do not particularly endorse one of another.  For demonstration purposes we will show using Gunicorn since it is a common choice in the industry.
+
+Previously we showed how to start the Guardrails API as a dev server using the `guardrails start` command.  When launching the Guardrails API with a WSGI server, you will reference the underlying `guardrails_api` module instead.  For example, when we Dockerize the Guardrails API for internal use, our final line is:
+```Dockerfile
+CMD gunicorn --bind 0.0.0.0:8000 --timeout=90 --threads=10 'guardrails_api.app:create_app(None, "config.py")'
+```
+
+This line starts the Guardrails API Flask application with a gunicorn WSGI server.  It specifies what port to bind the server to, as well as the timeout for workers and the maximum number of worker threads for handling requests.  We typically use the `gthread` worker class with gunicorn because of compatibility issues between how some async workers try to monkeypatch dependencies and how some libraries specify optional imports.
+
+Also note that we make the intentional decision to utilize threads over workers here for simple use-cases.  You could just as easily swap the `--threads` setting with the `--workers` setting above. The key tradeoff here is the impact that has on total resource consumption.  Since the `config.py` file is loaded at startup, running multiple workers means that each worker may need to load the models utilized by any validators in the config.  For use cases that have square-wave-like or sustained high traffic, this may be a tradeoff you want to make.
+
```

Comment:
please invert this. Let's give the example of multiple workers, and then discuss the tradeoffs of moving to mult threads. Let's also record the threading issue here

Reply:
Addressed in https://github.com/guardrails-ai/guardrails/pull/906